Source/DafnyCore/DooFile.cs

```diff
@@ -113,7 +118,7 @@ public class ManifestData {
     var tw = new StringWriter {
       NewLine = "\n"
     };
-    var pr = new Printer(tw, ProgramSerializationOptions, PrintModes.Serialization);
+    var pr = new Printer(tw, dafnyProgram.Options, PrintModes.Serialization);
```

Comment:
FYI I was able to (on a wip branch) cleanly fix https://github.com/dafny-lang/dafny/issues/4985 by just storing `function-syntax` in the doo file with a no-op check, because you already changed things so that the text in a doo file gets parsed downstream with the options from the doo file.

Reply:
Never mind, now I recall I had to make this change as well. :)